bot-itself/bot/commands.py

```diff
@@ -1,17 +1,124 @@
 import bot.answers as answers
 from models.chat import sync_chat
+from models.client import save, save_new_client, get_client_by_tg_id, get_client_by_id, get_client_by_username, get_wait_list
+
 logger.info("Starting bot ")
 
+
 @bot.message_handler(commands=['start'])
 def send_welcome(message):
-    sync_chat(message)
     logger.info(f'{message.from_user.username} sent {message.text}')
-    bot.send_message(message.chat.id, answers.start)
+    if message.chat.type == 'private':
+        client = get_client_by_tg_id(message.from_user.id)
+        if client is None:
+            save_new_client(message.from_user.id, message.from_user.username)
+            bot.send_message(message.chat.id, "Your request has been sent to the admin, wait for the approvement")
+        else:
+            if client.is_declined:
+                bot.send_message(message.chat.id, "Sorry, but your request was declined")
+            else:
+                if client.is_approved:
+                    bot.send_message(message.chat.id,
+                                     "Your request already has been approved. In order to get VPN credentials write "
+                                     "/get_creds")
+                else:
+                    bot.send_message(message.chat.id,
+                                     "You already requested VPN credentials. We will notify you as soon as answers "
+                                     "approves your request")
+def admin_only(func):
+    def wrapper(message):
+        client = get_client_by_tg_id(message.from_user.id)
+        if client is None or (not client.is_admin):
+            bot.send_message(message.chat.id, "You do not have admin rights to do it!")
+            return
+        return func(message)
+    return wrapper
+
+@bot.message_handler(commands=['waiting_list'])
+@admin_only
+def waiting_list(message):
+    logger.info(f'{message.from_user.username} sent {message.text}')
+    clients = get_wait_list()
+    clients_list = str()
+    number = 1
+    for client in clients:
+        clients_list += f"{number}. id: {client.id}, username: @{client.username}, telegram_id: {client.tg_id}\n"
+        number += 1
+    if clients_list == "":
+        clients_list = "There are no clients on the waiting list!"
+    bot.send_message(message.chat.id, clients_list)
+
+@bot.message_handler(commands=['approve'])
+@admin_only
+def client_approve(message):
+    logger.info(f'{message.from_user.username} sent {message.text}')
+    identity = message.text.split(' ')[1]
+    client = get_client_by_username(identity)
+    if client is None:
+        client = get_client_by_id(int(identity))
+    if client is None:
+        bot.send_message(message.chat.id, "This user hasn't been found!")
+        return
+    if client.is_approved:
+        bot.send_message(message.chat.id, "This user has already been approved!")
+    else:
+        client.is_approved = True
+        client.is_declined = False
+        save(client)
```

Comment:
probably either rename it function to like saveClient or smth. Or import it another way so u could use this function like `models.save`